### ISSUE TITLE
Question/Feature Request: Intermediate Results

### ISSUE THREAD

**Issue by Droggelbecher** [joblib/joblib#388]
I usually use joblib for long-running computations (as probably many people do).
In my particular set of applications the ultimate goal is often some kind of plot that shows results of all the jobs.

Now to judge whether my code produces what I want some small number of these outputs might suffice, and being able to look at those before Parallel() completed could often save a lot of time (I couldnt judge beforehand how many results I need to see something works good/bad though).

I saw on https://pythonhosted.org/joblib/parallel.html that this is nicely possible if jobs can be divided into groups, each of which is computed in parallel and report in between. However in my application these synchronization barriers would be devastating as jobs can have very different run times.

TL;DR:
Is there a not-too-complicated way to collect intermediate results in joblib without synchronization barriers? I don't care about order or timing of these results, I just want to be able to take a look every now and then at whats already there.
Or is this not nicely possible and I have to use multiprocessing (or whatever) by hand instead and roll my own communication?

Thanks in advance for any advice!

**Comment by eickenberg** [comment]
Couldn't you have your jobs write something to disk in a uniquely named
file and collect those from time to time?
If your jobs are long-running, chances are this won't affect performance.

**Comment by Droggelbecher** [comment]
Yeah thats kind-of what I'm doing right now. That is I produce human-readable files per job right now and were about to think about making them more machine-readable and duplicate the plot generation into something that collects info from these files, so performance-wise this is not a problem.

I was just hoping there would be a more straight-forward way, that doesnt require to encode/decode my info into files etc... but just use the plot function that I already have. But if there is no such thing, I'll happily go and do it via files.

**Comment by Droggelbecher** [comment]
Here I am one hour later with a neat solution using pickling, thanks for the hint!

**Comment by GaelVaroquaux** [comment]
Ideally, we'd love to have Parallel as an iterator, rather than a list,
as discussed a bit in
https://github.com/joblib/joblib/issues/79

It's a quite a lot of work, though.

I guess that I am closing the closing, as duplicate of #79, because we
need to clean up the list of open issues. It's not that we feel that it's
not important, though!

**Comment by GaelVaroquaux** [comment]
Well, it was already closed :).
